flink-connectors/flink-connector-aws-kinesis-data-streams/src/main/java/org/apache/flink/connector/kinesis/sink/KinesisDataStreamsSinkBuilder.java

```diff
@@ -0,0 +1,127 @@
+ *                     .setPartitionKeyGenerator(element -> String.valueOf(element.hashCode()))
+ *                     .build();
+ *
+ * KinesisDataStreamsSink<String> kdsSink =
+ *                 KinesisDataStreamsSink.<String>builder()
+ *                         .setElementConverter(elementConverter)
+ *                         .setStreamName("your_stream_name")
+ *                         .build();
+ * }</pre>
+ *
+ * <p>If the following parameters are not set in this builder, the following defaults will be used:
+ *
+ * <ul>
+ *   <li>{@code maxBatchSize} will be 200
+ *   <li>{@code maxInFlightRequests} will be 16
+ *   <li>{@code maxBufferedRequests} will be 10000
+ *   <li>{@code flushOnBufferSizeInBytes} will be 64MB i.e. {@code 64 * 1024 * 1024}
+ *   <li>{@code maxTimeInBufferMS} will be 5000ms
+ *   <li>{@code failOnError} will be false
+ * </ul>
+ *
+ * @param <InputT> type of elements that should be persisted in the destination
+ */
+@PublicEvolving
+public class KinesisDataStreamsSinkBuilder<InputT>
+        extends AsyncSinkBaseBuilder<
+                InputT, PutRecordsRequestEntry, KinesisDataStreamsSinkBuilder<InputT>> {
+
+    private static final int DEFAULT_MAX_BATCH_SIZE = 500;
+    private static final int DEFAULT_MAX_IN_FLIGHT_REQUESTS = 16;
+    private static final int DEFAULT_MAX_BUFFERED_REQUESTS = 10000;
+    private static final long DEFAULT_MAX_BATCH_SIZE_IN_B = 5 * 1024 * 1024;
+    private static final long DEFAULT_MAX_TIME_IN_BUFFER_MS = 5000;
+    private static final long DEFAULT_MAX_RECORD_SIZE_IN_B = 1 * 1024 * 1024;
+    private static final boolean DEFAULT_FAIL_ON_ERROR = false;
+
+    private Boolean failOnError;
+    private String streamName;
+    private Properties kinesisClientProperties;
+
+    KinesisDataStreamsSinkBuilder() {}
+
+    /**
+     * Sets the name of the KDS stream that the sink will connect to. There is no default for this
+     * parameter, therefore, this must be provided at sink creation time otherwise the build will
+     * fail.
+     *
+     * @param streamName the name of the stream
+     * @return {@link KinesisDataStreamsSinkBuilder} itself
+     */
+    public KinesisDataStreamsSinkBuilder<InputT> setStreamName(String streamName) {
+        this.streamName = streamName;
+        return this;
+    }
+
+    public KinesisDataStreamsSinkBuilder<InputT> setFailOnError(boolean failOnError) {
+        this.failOnError = failOnError;
+        return this;
+    }
+
+    public KinesisDataStreamsSinkBuilder<InputT> setKinesisClientProperties(
+            Properties kinesisClientProperties) {
+        this.kinesisClientProperties = kinesisClientProperties;
+        return this;
+    }
+
+    @Override
+    public KinesisDataStreamsSink<InputT> build() {
+        return new KinesisDataStreamsSink<>(
+                getElementConverter(),
+                getMaxBatchSize() == null ? DEFAULT_MAX_BATCH_SIZE : getMaxBatchSize(),
+                getMaxInFlightRequests() == null
```

Comment:
nit: You could use optionals here, this would remove duplicate method invocations

```
Optional.ofNullable(getMaxInFlightRequests()).orElse(DEFAULT_MAX_IN_FLIGHT_REQUESTS)
```